src/bootstrap/config.rs

```diff
@@ -86,9 +86,6 @@ pub struct Config {
     pub incremental: bool,
     pub dry_run: bool,
     /// `None` if we shouldn't download CI compiler artifacts, or the commit to download if we should.
-    #[cfg(not(test))]
-    download_rustc_commit: Option<String>,
-    #[cfg(test)]
     pub download_rustc_commit: Option<String>,
```

Comment:
please don't change this directly, use `config.download_rustc()` instead. The commit is lazily loaded and may be None even if we would end up downloading. If the problem is you need a full Builder, you might be able to move these functions to Build instead; I have a WIP for that in https://github.com/rust-lang/rust/pull/102282/commits/1c3575585e748c618f9e6fcc7ab42d2f18ffe70f if it seems like a lot of work, I can make a separate PR for it.

Reply:
Yeah, can you put it in a separate PR?

Reply:
https://github.com/rust-lang/rust/pull/104188

Reply:
Rebased on top of that, and stopped using the `Config` field directly.